iquip/apps/scheduler.py

```diff
@@ -0,0 +1,327 @@
+from PyQt5.QtWidgets import (
+    QStyleOptionViewItem, QWidget, QLayout, QLabel, QListView,
+    QPushButton, QHBoxLayout, QVBoxLayout, QAbstractItemDelegate, QAction
+)
+
+import qiwis
+from iquip.protocols import ExperimentInfo
+
+def _dismiss_items(layout: Optional[QLayout] = None):
+    """Auxiliary function for decoupling components in the layout.
+
+    Args:
+        layout: The layout whose elements are intended to be deleted.
+    """
+    if layout is not None:
+        while layout.count():
+            item = layout.takeAt(0)
+            widget = item.widget()
+            if widget is not None:
+                widget.setParent(None)
+                layout.removeWidget(widget)
+                widget.deleteLater()
+                widget = None
+            else:
+                _dismiss_items(item.layout())
+
+
+class SchedulerFrame(QWidget):
+    """Frame for displaying the submitted experiment list.
+    
+    Attributes:
+        expRun: The ExperimentView widget displaying currently running experiment.
+        expList: The QListView widget holding queued experiments.
+        model: The ExperimentModel that manages the data of expList.
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.expRun = RunningExperimentView()
+        self.expList = QListView()
+        self.expList.setItemDelegate(ExperimentDelegate(self.expList))
+        self.expList.setMovement(QListView.Free)
+        self.expList.setObjectName("Queued Experiments")
+        self.model = ExperimentModel([])
+        self.expList.setModel(self.model)
+        # layout
+        layout = QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 20)
+        layout.addWidget(QLabel("Currently running:"))
+        layout.addWidget(self.expRun)
+        layout.addWidget(QLabel("Queued experiments:"))
+        layout.addWidget(self.expList)
+
+
+class RunningExperimentView(QWidget):
+    """Widget for displaying the information of the experiment, especially for the one running.
+    
+    Attributes:
+        argsLayout: The HBoxLayout for displaying the experiment information besides its name.
+        nameLabel: The QLabel instance for displaying the experiment name.
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.nameLabel = QLabel("None", self)
+        self.args = []
+        # layout
+        layout = QVBoxLayout(self)
+        self.argsLayout = QHBoxLayout()
+        layout.addWidget(self.nameLabel)
+        layout.addLayout(self.argsLayout)
+
+    def updateInfo(self, info: Optional[ExperimentInfo] = None):
+        """Updates the information by modification.
+        This updates expRun when new experiment starts to run.
+        If info is None, it means there is no experiments running.
```

Comment:
Please write this line in the argument docstring for `info` since it describes the API specification of `info`.
In addition, "None if there is no experiments running." sounds more natural. Make docstring (especially for API specification) for users.